utils/proc/process.go

```diff
@@ -53,27 +53,30 @@ func FindProcess(ctx context.Context, pid int) (p IProcess, err error) {
 	return
 }
 
+// IsProcessRunning states whether a process is running or not. An error is returned if the context is Done while looking for the process state.
+func IsProcessRunning(ctx context.Context, pid int) (running bool, err error) {
+	err = parallelisation.DetermineContextError(ctx)
+	if err != nil {
+		return
+	}
+	p, SubErr := FindProcess(ctx, pid)
+	switch {
+	case SubErr == nil:
+		running = p.IsRunning()
```

Comment:
based on L223, does this return an error? Or should the one on L223 not return an error?

Or is it a different process type?

Reply:
I tried to describe it in the function comment but the error is only to describe whether the context was cancelled or timed out